docs/DeveloperGuide.md

```diff
@@ -238,7 +238,56 @@ _{more aspects and alternatives to be added}_
 
 _{Explain here how the data archiving feature will be implemented}_
 
+### Filter feature for transactions
 
+#### Implementation
+
+The filter transaction mechanism is facilitated by `FilterTransCommand` which extends from `Command` and `FilterTransCmdParser` which extends from `Parser`. To invoke the filter command, `FilterTransCmdParser` will parse the arguments from the user input via `FilterTransCmdParser#parse()` and returns the filter command if the arguments are valid.
+
+`FilterTransCommand` implements the `FilterTransCommand#execute()` operation which executes the command and returns the result message in a `CommandResult` object.
+
+The operation is exposed in the `Logic` interface as `Logic#execute()`.
+
+Given below is an example usage scenario and how the filter mechanism behaves at each step.
+
+Step 1. The user launches the application. The `UiManager` will call on the `MainWindow` to invoke the UI which displays the clients.
+
+![FilterTransState0](images/FilterTransState0png.png)
```

Comment:
Everything looks great! Perhaps renaming the png image file from FilterTransState0png.png to just FilterTransState0.png would be cleaner :)

Reply:
sure, thank you